src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -367,6 +367,28 @@ public GroupBy GroupBy(string columnName)
             DataFrameColumn column = _columnCollection[columnIndex];
             return column.GroupBy(columnIndex, this);
         }
+                
+        /// <summary>
+        /// Groups the rows of the <see cref="DataFrame"/> by unique values in the <paramref name="columnName"/> column.
+        /// </summary>
+        /// <typeparam name="TKey">Type of column used for grouping</typeparam>
+        /// <param name="columnName">The column used to group unique values</param>
+        /// <returns>A GroupBy object that stores the group information.</returns>
+        public GroupBy<TKey> GroupBy<TKey>(string columnName)
+        {
+            int columnIndex = _columnCollection.IndexOf(columnName);
```

Comment:
Can this call into the existing `GroupBy GroupBy(string columnName)` method above and then do the appropriate cast? That way we don't need to duplicate this code.